walkeeper/src/receive_wal.rs

```diff
@@ -186,37 +183,62 @@ fn request_callback(conf: WalAcceptorConf, timelineid: ZTimelineId, tenantid: ZT
     }
 }
 
-impl ReceiveWalConn {
-    pub fn new(socket: TcpStream, conf: WalAcceptorConf) -> Result<ReceiveWalConn> {
-        let peer_addr = socket.peer_addr()?;
-        let conn = ReceiveWalConn {
-            timeline: None,
-            stream_in: BufReader::new(socket.try_clone()?),
-            stream_out: socket,
+impl<'pg> ReceiveWalConn<'pg> {
+    pub fn new(pg: &'pg mut PostgresBackend) -> Result<ReceiveWalConn<'pg>> {
+        let peer_addr = pg.get_stream_in()?.get_ref().peer_addr()?;
```

Comment:
Not very important, but instead of making `get_stream_in` public only for this caller I'd rather add `get_peer_addr()` to `PostgresBackend`.

Reply:
Yup, that makes a bunch of sense! Fixed :)